src/ui/public/kuery/functions/utils/get_fields.js

```diff
@@ -6,13 +6,19 @@ export function getFields(node, indexPattern) {
     const fieldName = literal.toElasticsearchQuery(node);
     const field = indexPattern.fields.byName[fieldName];
     if (!field) {
-      throw new Error(`Field ${fieldName} does not exist in index pattern ${indexPattern.title}`);
+      return [{
```

Comment:
Maybe a comment in each of these to explain why we're doing this would be nice :)

Reply:
The spec for `Field` is actually a lot more involved than this:

![screenshot 2018-03-15 12 40 44](https://user-images.githubusercontent.com/404731/37487237-29130f18-284e-11e8-8f01-59acd0768278.png)

I realize this may be all the information you need, but I wonder if you should, or even can, create this field definition using the `Field` spec.

To Lukas's comment, adding a comment is fine, but making a little helper would be clear enough too:

```js
const createMissingField = name => [{ name, scripted: false }];

// then
return createMissingField(fieldName);
// and
return createMissingField(wildcard.toElasticsearchQuery(node));
```

Reply:
I thought about creating a full blown fake field but it seemed like overkill, but I agree this return value is a bit weird and fragile. Maybe instead of returning a fake field in this situation, I should just return an empty array and update is.js and range.js to use the user provided field name if no matches are found?

Reply:
> I thought about creating a full blown fake field but it seemed like overkill

True. Handling it upstream might be nicer. You could also just return some constant value indicating that the field isn't in the index pattern and then, as you say, let `is.js` and `range.js` react to that value.

```
import { FIELD_MISSING_FROM_INDEX_PATTERN } from 'path/to/constants';
return FIELD_MISSING_FROM_INDEX_PATTERN
```

That seems a little safer than just an empty array, and it's easier to understand what's happening by reading the code.

Reply:
What's unsafe about an empty array? I think it's pretty common for functions that return collections to return an empty collection if there are no matches (like Array.prototype.filter, for example). I like keeping the return type consistent so calling code doesn't have to have a bunch of conditionals.

Reply:
"Safe" was probably the wrong word to use. What I mean is, when you get an empty array back form the method, it's not really clear what that value means. Was there simply no match, or was the query on a field that doesn't exist?

If you actually want the empty array to indicate that, then fine, by all means do it that way. The only piece you lose is the "why," which perhaps you don't actually care about upstream.

Reply:
I'm confused, I'm not sure if we're talking about the same thing 😅. Pushed some code to hopefully clarify what I mean. @lukasolson I also added a comment :)

Let me know what you guys think

Reply:
:shipit: